docs/api/standard_library_types.md

```diff
@@ -774,356 +774,7 @@ In case you want to constrain the UUID version, you can check the following type
-)
-assert m.model_dump() == data
-```
-
-!!! note
-    Using the [`typing.Annotated` fields syntax](../concepts/json_schema.md#typingannotated-fields) can be handy to regroup
-    the `Union` and `discriminator` information. See the next example for more details.
-
-    There are a few ways to set a discriminator for a field, all varying slightly in syntax.
-
-    For `str` discriminators:
-    ```
-    some_field: Union[...] = Field(discriminator='my_discriminator'
-    some_field: Annotated[Union[...], Field(discriminator='my_discriminator')]
-    ```
-
-    For `CallableDiscriminator` discriminators:
-    ```
-    some_field: Union[...] = Field(discriminator=CallableDiscriminator(...))
-    some_field: Annotated[Union[...], CallableDiscriminator(...)]
-    some_field: Annotated[Union[...], Field(discriminator=CallableDiscriminator(...))]
-    ```
-
-!!! warning
-    Discriminated unions cannot be used with only a single variant, such as `Union[Cat]`.
-
-    Python changes `Union[T]` into `T` at interpretation time, so it is not possible for `pydantic` to
-    distinguish fields of `Union[T]` from `T`.
-
-#### Nested Discriminated Unions
-
-Only one discriminator can be set for a field but sometimes you want to combine multiple discriminators.
-You can do it by creating nested `Annotated` types, e.g.:
-
-```py requires="3.8"
-from typing import Literal, Union
-
-from typing_extensions import Annotated
-
-from pydantic import BaseModel, Field, ValidationError
-
-
-class BlackCat(BaseModel):
-    pet_type: Literal['cat']
-    color: Literal['black']
-    black_name: str
-
-
-class WhiteCat(BaseModel):
-    pet_type: Literal['cat']
-    color: Literal['white']
-    white_name: str
-
-
-Cat = Annotated[Union[BlackCat, WhiteCat], Field(discriminator='color')]
-
-
-class Dog(BaseModel):
-    pet_type: Literal['dog']
-    name: str
-
-
-Pet = Annotated[Union[Cat, Dog], Field(discriminator='pet_type')]
-
-
-class Model(BaseModel):
-    pet: Pet
-    n: int
-
-
-m = Model(pet={'pet_type': 'cat', 'color': 'black', 'black_name': 'felix'}, n=1)
-print(m)
-#> pet=BlackCat(pet_type='cat', color='black', black_name='felix') n=1
-try:
-    Model(pet={'pet_type': 'cat', 'color': 'red'}, n='1')
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for Model
-    pet.cat
-      Input tag 'red' found using 'color' does not match any of the expected tags: 'black', 'white' [type=union_tag_invalid, input_value={'pet_type': 'cat', 'color': 'red'}, input_type=dict]
-    """
-try:
-    Model(pet={'pet_type': 'cat', 'color': 'black'}, n='1')
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for Model
-    pet.cat.black.black_name
-      Field required [type=missing, input_value={'pet_type': 'cat', 'color': 'black'}, input_type=dict]
-    """
-```
+Pydantic has extensive support for union validation, both `typing.Union` and Python 3.10's pipe syntax (`A | B`) are supported.
```

Comment:
Could we add a link to the concepts/unions page here?

Reply:
oh yes, sorry that should be first.